security/pkg/nodeagent/secretfetcher/secretfetcher.go

```diff
@@ -443,6 +450,23 @@ func (sf *SecretFetcher) FindIngressGatewaySecret(key string) (secret model.Secr
 	val, exist := sf.secrets.Load(key)
 	secretFetcherLog.Debugf("load secret %s from secret fetcher: %v", key, exist)
 	if !exist {
+		// In some case, we might not get the secret from watch. Try to directly obtain the secret from API call
```

Comment:
This sounds like a potential kubernetes bug if its not included in the watch, is there an issue for this?

Reply:
+1, better to have more logs to prove watch didn't work(after k8s secret was created successfully).

Reply:
Watch could fail if we have an obsoleted watcher socket that were not closed promptly. We hit this quite often in GKE On-prem periodic tests which makes TLS failed. I agree it would be nice if watcher failure doesn't happen, but if it does or there is a time lag between watcher and secret request,  it is would be nice if we can remedy such failure, because otherwise we just failed and couldn't get out.

Reply:
or alternatively, return the secret and add to cache, however don't use ```sf.AddSecret(secret)``` to add to cache since this func is watcher callback that trigger push event; since you return the secret so push event shouldn't be triggered.
the logic would be simpler and less error-prone.